### PR TITLE
Make MVP UIs

### PR DESCRIPTION
**Pull request by menynexi**
add GetGameDialogue to update the gameStage.html page's dialogue using the PlayersDatabase (implemented in this commit : 77a502d) to access the Players currentPageID which will decide the level you play and the career path the player will follow. Additionaly using the GameStageDatabase (implemented in this commit : 3421029) to get the dialogue belonging to the currentPageID and display it to gameStage.html

add GetGameDialogueTest

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/GetGameDialogue.java [hunk 1]
```diff
@@ -24,21 +29,24 @@
...
 
-    Gson gson = new Gson();
-    String dialogue = gson.toJson(currentGameStage.getContent());
-
-    response.setContentType(JSON_CONTENT_TYPE);
-    response.getWriter().println(dialogue);
+      response.setContentType(JSON_CONTENT_TYPE);
+      response.getWriter().println(dialogue);
+    } catch (Exception nullPointer) {
+    }
```
Comment (lzoebaker) [inline src/main/java/com/google/sps/servlets/GetGameDialogue.java:50]: You should handle the exception is some way. Right now you are squashing the exception, which could be dangerous, see [resource here](https://riptutorial.com/java/example/15320/pitfall---ignoring-or-squashing-exceptions#:~:text=To%20make%20things%20worse%2C%20when,source%20of%20the%20problem%20was).
You could implement an error message being printed out to the screen, or even at least print out the error message to the console.

File: src/test/java/com/google/sps/GetGameDialogueTest.java [hunk 2]
```diff
@@ -1,56 +1,92 @@
...
+  private static String NAME_PROPERTY = "name";
+  private static String ID_PROPERTY = "id";
+  private static String CONTENT_PROPERTY = "content";
+  private static String QUIZ_KEY_PROPERTY = "quizkey";
+  private static String IS_LAST_STAGE_PROPERTY = "isFinalStage";
+  private static String NEXT_STAGE_PROPERTY = "nextstage";
+  private static String NAME_PLAYER = "many_admin";
+  private static String EMAIL_PLAYER = "admin@google.com";
+  private static String EXEPTION_PLAYER =
+      "The PlayerDatabase does not hold the players curentPageID";
```
Comment (lzoebaker) [inline src/test/java/com/google/sps/GetGameDialogueTest.java:48]: Spelling error, EXEPTION --> EXCEPTION

File: src/test/java/com/google/sps/GetGameDialogueTest.java [hunk 3]
```diff
@@ -1,56 +1,92 @@
...
   @Mock private HttpServletRequest request;
   @Mock private HttpServletResponse response;
-  private GetGameDialogue getGameDialogue = new GetGameDialogue();
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
   private static final Gson gson = new Gson();
-  private static final String TEST_STRING = "hello this is a test";
+  private static final String QUERY_FOR_GAMESTAGE_ENTITY = "gamestage";
+  private static final String SOFTWARE_ENGINEER_LEVEL1_NAME = "Software Engineer Level 1";
+  private static final String SOFTWARE_ENGINEER_INPUT = "Software Engineer";
```
Comment (lzoebaker) [inline src/test/java/com/google/sps/GetGameDialogueTest.java:30]: Not using these variables.

File: src/test/java/com/google/sps/GetGameDialogueTest.java [hunk 4]
```diff
@@ -1,56 +1,88 @@
...
+  private static final String TEST_CONTENT = "Some test content here.";
+  private static final Boolean IS_NOT_LAST_STAGE = false;
+  private static final Boolean IS_LAST_STAGE = true;
+  private static String IMAGE_PARAMETER = "image";
+  private static String NAME_PROPERTY = "name";
+  private static String ID_PROPERTY = "id";
+  private static String CONTENT_PROPERTY = "content";
+  private static String QUIZ_KEY_PROPERTY = "quizkey";
+  private static String IS_LAST_STAGE_PROPERTY = "isFinalStage";
+  private static String NEXT_STAGE_PROPERTY = "nextstage";
```
Comment (lzoebaker) [inline src/test/java/com/google/sps/GetGameDialogueTest.java:44]: Not using these in your code.

File: src/main/java/com/google/sps/servlets/GetGameDialogue.java (hunk 1 above)
Comment (menynexi) [reply src/main/java/com/google/sps/servlets/GetGameDialogue.java:50]: will do

File: src/main/java/com/google/sps/servlets/GetGameDialogue.java [hunk 5]
```diff
@@ -24,21 +29,24 @@
...
+  private final PlayerDatabase playerDatabase = new PlayerDatabase(datastore);
+  private final GameStageDatabase gameStageDatabase = new GameStageDatabase(datastore);
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
+      // you can only use the PlayerDatabase sucessfully if you are running this on a live server
+      // because it requeres you to log in and there is no way to mock a user
+      GameStage currentGameStage =
+          gameStageDatabase.getGameStage(playerDatabase.getEntityCurrentPageID());
```
Comment (WillHipschman) [inline src/main/java/com/google/sps/servlets/GetGameDialogue.java:44]: extra line

File: src/main/java/com/google/sps/servlets/GetGameDialogue.java [hunk 6]
```diff
@@ -24,21 +29,24 @@
...
+  private static final Gson gson = new Gson();
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private final PlayerDatabase playerDatabase = new PlayerDatabase(datastore);
+  private final GameStageDatabase gameStageDatabase = new GameStageDatabase(datastore);
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
+      // you can only use the PlayerDatabase sucessfully if you are running this on a live server
+      // because it requeres you to log in and there is no way to mock a user
```
Comment (WillHipschman) [inline src/main/java/com/google/sps/servlets/GetGameDialogue.java:42]: typo: requeres-->requires

File: src/main/java/com/google/sps/servlets/GetGameDialogue.java [hunk 7]
```diff
@@ -24,21 +29,24 @@
...
+      String dialogue = gson.toJson(currentGameStage.getContent());
 
-    Gson gson = new Gson();
-    String dialogue = gson.toJson(currentGameStage.getContent());
-
-    response.setContentType(JSON_CONTENT_TYPE);
-    response.getWriter().println(dialogue);
+      response.setContentType(JSON_CONTENT_TYPE);
+      response.getWriter().println(dialogue);
+    } catch (Exception nullPointer) {
```
Comment (WillHipschman) [inline src/main/java/com/google/sps/servlets/GetGameDialogue.java:49]: Why do we ignore the exception? Can we catch a more specific exception?

File: src/test/java/com/google/sps/GetGameDialogueTest.java [hunk 8]
```diff
@@ -1,56 +1,79 @@
...
-
-    // mocks the result of querying the GetGameDialogue
-    this.getGameDialogue.doGet(this.request, this.response);
-    // checks that the string writer used in servlet mock response contains the database object JSON
-    // that matches with the hardcoded CareerQAndChoice given be the mock database
-    Gson gson = new Gson();
-    JsonElement expected = JsonParser.parseString(gson.toJson(TEST_STRING));
-    JsonElement result = JsonParser.parseString(stringWriter.toString());
-    Assert.assertEquals(expected, result);
+  public void getGameStageFromPlayerID_OutputsworksSuccessfully() throws Exception {
```
Comment (WillHipschman) [inline src/test/java/com/google/sps/GetGameDialogueTest.java:57]: name you tests method_condition_result

so doGet_gameStageFromPlayerId_succeeds

File: src/main/java/com/google/sps/servlets/GetGameDialogue.java (hunk 7 above)
Comment (DirtGrubDylan) [reply src/main/java/com/google/sps/servlets/GetGameDialogue.java:49]: +1

File: src/test/java/com/google/sps/GetGameDialogueTest.java (hunk 8 above)
Comment (DirtGrubDylan) [reply src/test/java/com/google/sps/GetGameDialogueTest.java:57]: Can we also get a test that covers the unsuccessful path?

**Pull request closed without merging**